contracts/src/registry/ETHRegistrar.sol

```diff
@@ -142,24 +142,20 @@ contract ETHRegistrar is IETHRegistrar, AccessControl {
         bytes32 secret,
         IRegistry subregistry,
         address resolver,
-        uint96 flags,
         uint64 duration
     ) external payable returns (uint256 tokenId) {
-        IPriceOracle.Price memory price = rentPrice(name, duration);
-        uint256 totalPrice = price.base + price.premium;
-        if (msg.value < totalPrice) {
-            revert InsufficientValue(totalPrice, msg.value);
-        }
+        uint256 totalPrice = checkPrice(name, duration);
 
-        _consumeCommitment(name, duration, makeCommitment(name, owner, secret, address(subregistry), resolver, flags, duration));
+        _consumeCommitment(name, duration, makeCommitment(name, owner, secret, address(subregistry), resolver, duration));
 
-        tokenId = registry.register(name, owner, subregistry, resolver, flags, uint64(block.timestamp) + duration);
+        uint64 expiry = uint64(block.timestamp) + duration;
+        tokenId = registry.register(name, owner, subregistry, resolver, ALL_ROLES, expiry);
```

Comment:
This needs adjusting - as it stands it will allow the user to renew their own name directly. Either this should mask out `ROLE_RENEW` and `ROLE_RENEW_ADMIN` or it should be set to only the safe roles.